iib/workers/tasks/build_containerized_merge.py

```diff
@@ -0,0 +1,376 @@
+        # Setup and clone Git repository
+        branch = prebuild_info['ocp_version']
+        (
+            index_git_repo,
+            local_git_repo_path,
+            localized_git_catalog_path,
+        ) = prepare_git_repository_for_build(
+            request_id=request_id,
+            from_index=source_from_index,
+            temp_dir=temp_dir,
+            branch=branch,
+            index_to_gitlab_push_map=index_to_gitlab_push_map or {},
+        )
+
+        # Pull both source and target index.db artifacts and read present bundle
+        source_index_db_path = fetch_and_verify_index_db_artifact(
+            source_from_index_resolved, temp_dir
+        )
+        target_index_db_path = fetch_and_verify_index_db_artifact(target_index_resolved, temp_dir)
+
+        # Get the bundles from the index.db file
+        with set_registry_token(overwrite_target_index_token, target_index, append=True):
+            source_index_bundles, source_index_bundles_pull_spec = _get_present_bundles(
+                source_index_db_path, temp_dir
+            )
+            log.debug("Source index bundles %s", source_index_bundles)
+            log.debug("Source index bundles pull spec %s", source_index_bundles_pull_spec)
+
+            target_index_bundles, target_index_bundles_pull_spec = _get_present_bundles(
+                target_index_db_path, temp_dir
+            )
+            log.debug("Target index bundles %s", target_index_bundles)
+            log.debug("Target index bundles pull spec %s", target_index_bundles_pull_spec)
+
+        # Validate the bundles from source and target have their pullspecs present in the registry
+        set_request_state(
+            request_id,
+            'in_progress',
+            'Validating whether the bundles have their pullspecs present in the registry',
+        )
+        unique_bundles = list(set(source_index_bundles + target_index_bundles))
+        validate_bundles_in_parallel(
+            bundles=unique_bundles,
+            threads=parallel_threads,
+            wait=True,
+        )
+
+        set_request_state(request_id, 'in_progress', 'Adding bundles missing in source index image')
+        log.info('Adding bundles from target index image which are missing from source index image')
+
+        missing_bundles, invalid_bundles = get_missing_bundles_from_target_to_source(
+            source_index_bundles=source_index_bundles,
+            target_index_bundles=target_index_bundles,
+            source_from_index=source_from_index_resolved,
+            ocp_version=prebuild_info['target_ocp_version'],
+            target_index=target_index_resolved,
+            ignore_bundle_ocp_version=ignore_bundle_ocp_version,
+        )
+
+        missing_bundle_paths = [bundle['bundlePath'] for bundle in missing_bundles]
+        if missing_bundle_paths:
+            add_max_ocp_version_property(missing_bundle_paths, temp_dir)
```

Comment:
while talking to Jan, we came across a comment which said this was done for v4.9. Since we don't support it in the new workflow, we can get rid of it.

https://github.com/release-engineering/iib/blob/master/iib/workers/tasks/build.py#L930